integration_tests/test_aggregator_batch_commands.py

```diff
@@ -28,31 +30,146 @@ def _setup(self):
-                logging.warning(json_grid_tree)
-            if "available_energy_kWh" in device_event["device_info"] and \
-                    device_event["device_info"]["available_energy_kWh"] > 0.0:
-                self.add_to_batch_commands.offer_energy(area_uuid=device_event["area_uuid"], price=1,
-                                                        energy=device_event["device_info"]["available_energy_kWh"] / 2) \
-                    .list_offers(area_uuid=device_event["area_uuid"])
-
-                if "energy_requirement_kWh" in device_event["device_info"] and \
-                        device_event["device_info"]["energy_requirement_kWh"] > 0.0:
-                    self.add_to_batch_commands.bid_energy(area_uuid=device_event["area_uuid"], price=30,
-                                                          energy=device_event["device_info"][
-                                                                     "energy_requirement_kWh"] / 2) \
-                        .list_bids(area_uuid=device_event["area_uuid"]) \
-                        .last_market_stats(area_uuid=device_event["area_uuid"])
-
-        if self.commands_buffer_length:
-            transaction = self.execute_batch_commands()
-            if transaction is None:
-                self.errors += 1
-            logging.info(f"Batch command placed on the new market")
+        try:
+            for device_event in market_info["content"]:
+                if "device_info" not in device_event or device_event["device_info"] is None:
+                    continue
+                if key_in_dict_and_not_none(device_event, "grid_stats_tree"):
+                    json_grid_tree = json.dumps(device_event["grid_stats_tree"], indent=2)
+                    logging.warning(json_grid_tree)
+
+                if self._can_place_offer(device_event):
+                    self.add_to_batch_commands.offer_energy(
+                        area_uuid=device_event['area_uuid'],
+                        price=1,
+                        energy=device_event['device_info']['available_energy_kWh'] / 4,
+                        replace_existing=False
+                    ).offer_energy(
+                        area_uuid=device_event['area_uuid'],
+                        price=1,
+                        energy=device_event['device_info']['available_energy_kWh'] / 4,
+                        replace_existing=False
+                    ).offer_energy(
+                        area_uuid=device_event['area_uuid'],
+                        price=1,
+                        energy=device_event['device_info']['available_energy_kWh'] / 4,
+                        replace_existing=True
+                    ).offer_energy(
+                        area_uuid=device_event['area_uuid'],
+                        price=1,
+                        energy=device_event['device_info']['available_energy_kWh'] / 4,
+                        replace_existing=False
```

Comment:
Rather than replacing bids with identical price and energy is to have different and validate in integration test that the last one was traded

Reply:
This is what I was trying to do [here](https://github.com/gridsingularity/d3a-api-client/pull/89/files/a5afed4a333d0c68d3a112fe4eef2e9b91f12370#diff-69601bf521aa6db62641cf28c676950ab0bd31b13024830a70d0c37b1e0df234R115-R117). But you might be right, manually checking the values could actually be more useful.

Reply:
Done!